cheroot/connections.py

```diff
@@ -155,15 +158,29 @@ def get_conn(self, server_socket):
 
             # Wait for the next tick to occur.
             return None
-
-        try:
-            # See if we have a new connection coming in.
-            rlist.remove(ss_fileno)
-        except ValueError:
-            # No new connection, but reuse existing socket.
-            conn = socket_dict[rlist.pop()]
-        else:
-            conn = server_socket
+        finally:
+            for sock in socket_dict:
+                self._selector.unregister(sock)
+
+        # See if we have a new connection coming in.
+        new_connection = False
+        rlist = []
+        for key, event in key_events:
+            if key.fd == ss_fileno:
+                # If we have a new connection, reuse the server socket
+                new_connection = True
+                conn = server_socket
+            else:
+                # Make a list of all the other sockets ready to read
+                rlist.append(key.fd)
+
+        if new_connection is False:
+            if len(rlist) == 0:
+                # If we didn't get any readable sockets, wait for the next tick
+                return None
+            else:
+                # No new connection, but reuse existing socket.
+                conn = socket_dict[rlist.pop()]
```

Comment:
Looking closer at this, I see that it's probably more readable to follow [EAFP](https://stackoverflow.com/a/11360880/595220)
```suggestion
            try:
                # No new connection, but reuse the existing socket.
                conn = socket_dict[rlist.pop()]
            except KeyError:
                raise  # fd not in the dict?
            except IndexError:
                # If we didn't get any readable sockets, wait for the next tick
                return None
```